pkg/vpcmodel/abstractVPC.go

```diff
@@ -40,10 +42,21 @@ type VPCResource struct {
 	VPCRef VPCResourceIntf `json:"-"`
 }
 
+func (n *VPCResource) ExtendedPrefix(c *VPCConfig) string {
+	if c.IsMultipleVPCsConfig {
+		return n.VPC().Name() + deliminator
+	}
+	return ""
+}
+
 func (n *VPCResource) Name() string {
 	return n.ResourceName
 }
 
+func (n *VPCResource) ExtendedName(c *VPCConfig) string {
+	return n.ExtendedPrefix(c) + n.Name()
+}
+
```

Comment:
why is `ExtendedName` defined in the level of `VPCResource` ?  seems it is only used in the level of `Node` and of `EndpointElem`  and Subnet ?
could be added only to `InternalNode` and `NodeSet` interfaces?

Reply:
For consistency, I defined it in the same level as the one Name() is defined as.
Moreover, `ExtendedName(..)` is called for objects that can potentially be `ExternalNetwork`. Defining `ExternalName` only for `InternalNode` and `NodeSet` interfaces will require checking in each such place what is the underlying type of the object and then calling either `Name()` or `ExtendedName(...)` (seems the best way of doing it is defining another member function of `VPCResourceIntf`). So all together we will have to maintain more (and I believe messier) code than just defining `ExtendedName(..)` for `ExternalNetwork`.

Reply:
but isn't `ExtendedName` relevant only for endpoints in multi-vpc context? (and not any vpc resource?)
seems if it's defined in the level of `Node` and `NodeSet` it may be sufficient and not require conversions?

Reply:
Yes, this should work. Just note that in this case ExtendedName will still need to be defined for ExternalNetwork.  Code lines will hardy be reduced; but I guess it makes more sense design wise.

Reply:
No, it does not work since `grouping` and `jsonOutput` uses `VPCResourceIntf` interface which can be either `Node` or `EndpointElem`
This change will require another abstract type - I doubt it is worth it. 

I removed the implementation of `ExtendedName()` from `nacl` and `sg` where it was not really needed, though.